### PR TITLE
ci: only run commit lint when only changing docs folder

### PR DESCRIPTION
**Pull request by mapleeit**
Two main changes in this PR:

1. Don't run `core-test` or any other test when just changing the docs. This will save a lot of time and resources.
2. Remove *adding-comment when commit lint fails* function because it won't work if the PR is raised by external contributors(in other words, the source branch of the PR comes from a forked repo). That means the value is very limited, and considering the complexity it adds, I would say it's a nice move to remove this functionality. So that this change won't increase a lot modulization concerns, because they are so simple.

### CODE REVIEW

File: .github/workflows/ci-no-code-related.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: CI for Code Un-related Changes
```
Comment (JoanFM) [inline .github/workflows/ci-no-code-related.yml:1]: why is this needed?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -2,6 +2,9 @@ name: CI
 
 on:
   pull_request:
+    paths-ignore:
+      - 'docs/**'
+      - '.github/workflows/**'
```
Comment (JoanFM) [inline .github/workflows/ci.yml:7]: github workflows should not be added here, changes here may be potential problems

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-no-code-related.yml (hunk 1 above)
Comment (mapleeit) [reply .github/workflows/ci-no-code-related.yml:1]: Commit lint is still needed for the code un-related part.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (mapleeit) [reply .github/workflows/ci.yml:7]: Ah. I see. For example, change the test logic in workflows. I'll remove this line.

File: .github/workflows/ci-no-code-related.yml (hunk 1 above)
Comment (JoanFM) [reply .github/workflows/ci-no-code-related.yml:1]: Can the `paths` be filtered at each step of the `regular CI`? Our CI is already non-modular enough

**Review by JoanFM — DISMISSED** [review]

File: .github/workflows/ci-no-code-related.yml (hunk 1 above)
Comment (mapleeit) [reply .github/workflows/ci-no-code-related.yml:1]: `paths` can not be used on step or job level. For the modular concerns, I have some ideas that can improve it. 

I'll push some changes later.

Comment (mapleeit) [reply .github/workflows/ci-no-code-related.yml:1]: I did some experiment. Unfortunately, [composite actions](https://docs.github.com/cn/actions/creating-actions/creating-a-composite-action) can't support all the function we need for CI moderation here.

**Review by hanxiao — CHANGES_REQUESTED** [review]
in the new `.github/workflows/ci-no-code-related.yml`, the netlify preview should still be built

**Comment by mapleeit** [comment]
@hanxiao Doesn't deploying stay in `.github/workflows/label-pr.yml`? It won't be affected.

**Comment by hanxiao** [comment]
oh good point
